packages/charts/src/utils.ts

```diff
@@ -210,3 +218,42 @@ export const generateTicks = ({
 
   return ticks;
 };
+
+export const calculateColorScale = (
+  minValue: number,
+  maxValue: number,
+  colorsArray?: string[],
+  colorMode: ColorModeType = 'shades'
+) => {
+  const colors =
+    colorsArray && colorsArray.length
+      ? colorsArray.length > 1
```

Comment:
For better readability - please remove multiple ternary operator and use default function argument. 
```
export const calculateColorScale = (
  colors = theme.colors,
) => {
 const [firstColor] = colors; // check for 'undefined'
 ...
}
```